### PR TITLE
add key and questKey models and seeds files

### PR DESCRIPTION
**Pull request by rosiene**

### CODE REVIEW

File: spec/models/quest_key_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe QuestKey, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (hezus) [inline spec/models/quest_key_spec.rb]: Please fix the tests

File: NarniaRules.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+- A bonus can have a maximum of +3 bonus. 
+
+The character during the game
+- During the game the story (quest) can ADD extra bonus points to a skill
+- During the game the story (quest) can SUBRTACT extra bonus points to a skill
+
+### Key's
+Key's are actions, events, roads, objects, choices that happens during that particulary piece of story (or as we call it quests)
+- Behind the scenes we create a list of these key's as the player goes through his choices. 
+- Key's can trigger **reciving** objects that goes in the bag.
```
Comment (wearethefoos) [inline NarniaRules.md]: receiving

Comment (wearethefoos) [inline NarniaRules.md]: Rename objects to Items, as the model is called to make it more clear.

File: app/models/key.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class Key < ActiveRecord::Base
+  validates :number, presence: true
```
Comment (wearethefoos) [inline app/models/key.rb]: Shouldn't this also be unique?

File: app/models/quest_key.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class QuestKey < ActiveRecord::Base
```
Comment (wearethefoos) [inline app/models/quest_key.rb:1]: Why is this a separate model?

File: db/keys/quest_key.rb [hunk 5]
```diff
@@ -0,0 +1,381 @@
+QuestKey.create(quest: Quest.find_by(ref_number: 137), key: Key.find_by(number: 1), event: "check" );
```
Comment (wearethefoos) [inline db/keys/quest_key.rb]: Maybe make class methods to find Quests and Keys by their numbers? E.g.:

``` ruby
Quest.by_ref(137)
Key.by_ref(1)
```

File: db/migrate/20160707141852_create_keys.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class CreateKeys < ActiveRecord::Migration
+  def change
+    create_table :keys do |t|
+      t.integer :number
```
Comment (wearethefoos) [inline db/migrate/20160707141852_create_keys.rb:4]: Would have been better to use the `ref_number` like we used in Quest

File: db/keys/quest_key.rb [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+QuestKey.create(
+  key: Key.find_by(number: 11),
+  quest:Quest.find_by(ref_number: 199),
+  choice: Choice.find_by(quest: Quest.find_by(ref_number: 199), destination: 274),
+  event: "checked"
+);
+
+QuestKey.create(
+  key: Key.find_by(number: 11),
+  quest:Quest.find_by(ref_number: 205),
```
Comment (wearethefoos) [inline db/keys/quest_key.rb]: space between quest: and Quest, it's everywhere..

File: app/models/quest_key.rb (hunk 4 above)
Comment (wearethefoos) [inline app/models/quest_key.rb:1]: Ah I see it now. A key has_and_belongs_to_many quests etc.

ok, all fine! :)

File: db/seeds.rb [hunk 8]
```diff
@@ -7,7 +7,11 @@
...
 Quest.destroy_all
+Key.destroy_all
+QuestKey.destroy_all
 
 Dir[Rails.root.join('db/stories/**/*.rb')].each { |f| require f }
+Dir[Rails.root.join('db/keys/key.rb')].each { |f| require f }
+Dir[Rails.root.join('db/keys/quest_key.rb')].each { |f| require f }
 
-puts "Seeded #{Quest.count} Quests and #{Choice.count} Choices!"
+puts "Seeded #{Quest.count} Quests, #{Choice.count} Choices, #{Key.count} Keys and #{QuestKey.count} QuestChoiceKey!"
```
Comment (wearethefoos) [inline db/seeds.rb]: QuestChoiceKey??

File: spec/factories/keys.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
+  factory :key do
+    number 1
```
Comment (wearethefoos) [inline spec/factories/keys.rb]: Make this a sequence. See: https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md#sequences

``` ruby
sequence :number { |n| n }
```

File: spec/factories/quest_keys.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :quest_key do
+    quest ""
+    key ""
+    event "MyString"
```
Comment (wearethefoos) [inline spec/factories/quest_keys.rb]: This is not a valid event. Create different factories for different events (aka `traits`) and test accordingly.

File: spec/models/key_spec.rb [hunk 11]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe Key, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (wearethefoos) [inline spec/models/key_spec.rb]: Remove line 4-8

File: spec/models/quest_key_spec.rb (hunk 1 above)
Comment (wearethefoos) [inline spec/models/quest_key_spec.rb]: Remove line 4-7

File: spec/models/quest_key_spec.rb [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+  describe "validation" do
+    it "is invalid without a key" do
+      key = Key.new(number: "")
+      key.valid?
+      expect(key.errors).to have_key(:number)
+    end
+  end
+end
+
+describe Quest do
```
Comment (wearethefoos) [inline spec/models/quest_key_spec.rb]: Move to Quest spec!

File: spec/models/quest_key_spec.rb [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  describe "validation" do
+    it "is invalid without a ref_number" do
+      quest = Quest.new(ref_number: "")
+      quest.valid?
+      expect(quest.errors).to have_key(:ref_number)
+    end
+  end
+end
+
+describe QuestKey do
```
Comment (wearethefoos) [inline spec/models/quest_key_spec.rb]: Remove this, and add to previous describe block for quest key spec.

File: app/models/key.rb [hunk 14]
```diff
@@ -0,0 +1,8 @@
+class Key < ActiveRecord::Base
+  validates :ref_number, presence: true
+  validates_uniqueness_of :ref_number
+
+  def self.by_ref(ref_number)
+    self.find_by(ref_number: ref_number)
```
Comment (wearethefoos) [inline app/models/key.rb]: no need for the `self.` here :)

File: app/models/quest.rb [hunk 15]
```diff
@@ -3,4 +3,8 @@ class Quest < ActiveRecord::Base
 
   validates_presence_of :description, :ref_number
   validates_uniqueness_of :ref_number
+
+  def self.by_ref(number)
+    self.find_by(ref_number: number)
```
Comment (wearethefoos) [inline app/models/quest.rb]: no need for the `self.`... :)

**Comment by wearethefoos** [comment]
Minor thingy with the `self.`

Other than that good! Let me know when it's fixed, and update the branch on the latest master (or click the button "Update branch" below).

**Comment by wearethefoos** [comment]
Nice, shipping this now.

**Pull request merged**